### PR TITLE
Fix the first argument of the `onChange` of the `RangePicker` when values are cleared

### LINKED ISSUE #757: the parameter of `onChange` is wrong when I click cancel button in `RangePicker`
# Production like

https://codesandbox.io/p/sandbox/rc-range-picker-onclick-fg9q9p?file=%2Fsrc%2FApp.tsx%3A1%2C1-21%2C1

### Steps to reproduce

1. open the reproduction
1. open dev tool
1. fill the dates
1. click the cancel button (icon with "x")
1. see the console

### What is expected?

The first parameter of `onChange` and `onCalendarChange` should be same.
I'm not sure how it is actually, but I think `[null, null]` makes scence.

[In the document](https://www.npmjs.com/package/rc-picker#rangepicker), the type of the `onChange` should be `(value: [moment], formatString: [string, string]) => void` (in the docs, it is written as `Function(value:[moment], formatString: [string, string])`). `value` can't be `null`.

### What is actually happening?

The first parameter of `onChange` and `onCalendarChange` are different.
`onChange` ... `null`
`onCalendarChange` ... `[null, null]`

| Environment | Info |
| --- | --- |
| rc-picker | 4.1.1 |
| React | 18.2.0 |
| System | macOS 14.0 |
| Browser | Google Chrome Version 121.0.6167.139 |

### PR DESCRIPTION
**Pull request by tnyo43**
close #757

I fix the first argument of the `onChange` of the `RangePicker` when values are cleared.
Currently, when I clear the values of the `RangePicker`, the first argument of the `onChange` is `null`.
The type of the `onChange` is `((dates: NoUndefinedRangeValueType<Date>, dateStrings: [string, string]) => void) | undefined`, so the first argument can't be `null`.

On the other hand, that of the `onCalendarChange` function is `[null, null]` and it makes sence if the `onChange` is so.

### CODE REVIEW

File: src/PickerInput/hooks/useRangeValue.ts [hunk 1]
```diff
@@ -289,11 +289,7 @@ export default function useRangeValue<ValueType extends DateType[], DateType ext
 
       // Trigger `onChange` if needed
       if (onChange && !isSameMergedDates) {
-        onChange(
-          // Return null directly if all date are empty
-          isNullValue && clone.every((val) => !val) ? null : clone,
-          getDateTexts(clone),
-        );
+        onChange(clone, getDateTexts(clone));
```
Comment (tnyo43) [inline src/PickerInput/hooks/useRangeValue.ts:292]: `onChange` is used as same as `onCalendarChange` is.

https://github.com/react-component/picker/blob/8bfd695ad3e2fbc9da09614c9b79de48bf032af1/src/PickerInput/hooks/useRangeValue.ts#L152-L154

File: src/PickerInput/hooks/useRangeValue.ts [hunk 2]
```diff
@@ -289,11 +289,7 @@ export default function useRangeValue<ValueType extends DateType[], DateType ext
 
       // Trigger `onChange` if needed
       if (onChange && !isSameMergedDates) {
-        onChange(
-          // Return null directly if all date are empty
```
Comment (tnyo43) [inline src/PickerInput/hooks/useRangeValue.ts:293]: It was difficult for me to understand why such specifications were made 😢 
I appreciate if someone tells me. If my change is not good for that, I also want to find better solution!

**Comment by tnyo43** [comment]
@zombieJ @li-jia-nan @MadCcc  Could you take a look and help move this improvement forward?

**Comment by zombieJ** [comment]
Sorry for delay. It's by design. `onChange` is not same as `onCalendarChange`.

ref: https://github.com/react-component/picker/issues/757#issuecomment-1965675890

**Comment by tnyo43** [comment]
> It's by design.

Ok, if that's the case, we still have an issue with the types. I'm thinking of addressing it in another PR, so I would appreciate it if you could check it out.

**Pull request closed without merging**